src/System.Runtime.Caching/src/System/Runtime/Caching/MemoryMonitor.Windows.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Specialized;
+using System.Security;
+using System.Runtime.InteropServices;
+
+
+namespace System.Runtime.Caching
+{
+    internal abstract partial class MemoryMonitor
+    {
+        static MemoryMonitor()
+        {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
```

Comment:
You don't need this check anymore as this file is only included on Windows.

Reply:
Thanks for catching this @safern!